zkstack_cli/crates/zkstack/src/commands/dev/commands/lint.rs

```diff
@@ -70,13 +72,18 @@ fn lint_rs(shell: &Shell, ecosystem: &EcosystemConfig, check: bool) -> anyhow::R
     let link_to_code = &ecosystem.link_to_code;
     let lint_to_prover = &ecosystem.link_to_code.join("prover");
     let link_to_zkstack = &ecosystem.link_to_code.join("zkstack_cli");
-    let paths = vec![link_to_code, lint_to_prover, link_to_zkstack];
 
     spinner.freeze();
-    for path in paths {
+    for path in [link_to_code, lint_to_prover, link_to_zkstack] {
         let _dir_guard = shell.push_dir(path);
         let mut cmd = cmd!(shell, "cargo clippy");
-        let common_args = &["--locked", "--", "-D", "warnings"];
+        let mut common_args = vec!["--locked", "--", "-D", "warnings"];
+
+        if path.ends_with("zkstack_cli") {
```

Comment:
seems we missed this part for core components as well. only inside prover it's allowed